### ISSUE TITLE
Chrome debug protocol missing api: beforeExecuting event

### ISSUE THREAD

**Issue by Alphapage** [Microsoft/ChakraCore#4253]
Hello,

You have implemented the chrome debug protocol and it works without problem 👍 
But this protocol is missing some features that are necessary to reach all debugger capacity.

I'm trying to debug a typescript project, but I can't without the addition of an `outFiles` property in vscode `launch.json` which is an unnecessary overhead (in my opinion)
We just need a new event callback which will be raised before the execution of each script including `eval` ones to be able to resolve sourcemaps, map the authored source, then set breakpoints to the script before starting execution.

I created an issue in vscode-chrome-debug-core https://github.com/Microsoft/vscode-chrome-debug-core/issues/255 : it is a nodejs app with 2 files which shows the problem.

It seems chromium isn't planning such an api, so I just ask if you want to enable such a feature in chakracore because you provide innovative debugger api like time travel debugging...

Thank you in advance for your help.

**Comment by digitalinfinity** [comment]
Hi @Alphapage- are you using node-chakracore?

**Comment by Alphapage** [comment]
Hi @digitalinfinity .
For my example, I use node-chakracore, but chrome debugger protocol is part of chakracore, so it should be the same for chakracore browser (don't remember the name ;-) but it hides somewhere on the _edge_ ).
Maybe node-chakracore already implements such a feature ?

**Comment by digitalinfinity** [comment]
Hi @Alphapage - actually, CDP is implemented in node-chakracore as part of the chakrashim, not in chakracore 😄 - can you file an issue there, we'll be happy to discuss there!

**Comment by Alphapage** [comment]
I posted a new issue in node-chakracore, but I think chackracore should provide chrome debugger protocol shims + additional features too.
This could solve some limitations and unify debugging tools like vscode-chrome-debug, vscode-node-debug or vscode-edge-debug.
This is just my opinion.

**Comment by digitalinfinity** [comment]
Thanks @Alphapage- this is an interesting idea and something that we've talked about internally in the past- we'll let you know if there is an update here. Thanks for your feedback!
